src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1251,50 +1279,56 @@ internal static bool TryFormatTimeOnlyR(int hour, int minute, int second, Span<c
         //   012345678901234567890123456789012
         //   ---------------------------------
         //   2017-06-12
-        internal static bool TryFormatDateOnlyO(int year, int month, int day, Span<char> destination)
+        internal static bool TryFormatDateOnlyO<TChar>(int year, int month, int day, Span<TChar> destination) where TChar : unmanaged, IBinaryInteger<TChar>
         {
             if (destination.Length < 10)
             {
                 return false;
             }
 
-            WriteFourDecimalDigits((uint)year, destination, 0);
-            destination[4] = '-';
-            WriteTwoDecimalDigits((uint)month, destination, 5);
-            destination[7] = '-';
-            WriteTwoDecimalDigits((uint)day, destination, 8);
+            FormattingHelpers.WriteFourDecimalDigits((uint)year, destination, 0);
+            destination[4] = TChar.CreateTruncating('-');
+            FormattingHelpers.WriteTwoDecimalDigits((uint)month, destination, 5);
+            destination[7] = TChar.CreateTruncating('-');
+            FormattingHelpers.WriteTwoDecimalDigits((uint)day, destination, 8);
             return true;
         }
 
         // Rfc1123
         //   01234567890123456789012345678
         //   -----------------------------
         //   Tue, 03 Jan 2017
-        internal static bool TryFormatDateOnlyR(DayOfWeek dayOfWeek, int year, int month, int day, Span<char> destination)
+        internal static bool TryFormatDateOnlyR<TChar>(DayOfWeek dayOfWeek, int year, int month, int day, Span<TChar> destination) where TChar : unmanaged, IBinaryInteger<TChar>
         {
             if (destination.Length < 16)
             {
                 return false;
             }
 
-            string dayAbbrev = InvariantAbbreviatedDayNames[(int)dayOfWeek];
-            Debug.Assert(dayAbbrev.Length == 3);
-
-            string monthAbbrev = InvariantAbbreviatedMonthNames[month - 1];
-            Debug.Assert(monthAbbrev.Length == 3);
-
-            destination[0] = dayAbbrev[0];
-            destination[1] = dayAbbrev[1];
-            destination[2] = dayAbbrev[2];
-            destination[3] = ',';
-            destination[4] = ' ';
-            WriteTwoDecimalDigits((uint)day, destination, 5);
-            destination[7] = ' ';
-            destination[8] = monthAbbrev[0];
-            destination[9] = monthAbbrev[1];
-            destination[10] = monthAbbrev[2];
-            destination[11] = ' ';
-            WriteFourDecimalDigits((uint)year, destination, 12);
+            if (typeof(TChar) == typeof(char))
+            {
+                Span<char> dest = MemoryMarshal.Cast<TChar, char>(destination);
+
+                FormattingHelpers.CopyFourChars("Sun,Mon,Tue,Wed,Thu,Fri,Sat,".AsSpan(4 * (int)dayOfWeek), dest);
```

Comment:
>dayOfWeek [](http://example.com/codeflow?start=95&length=9)

worth asserting the value of dayOfWeek. I know the original code was not doing it.